api/src/main/java/org/apache/iceberg/variants/VariantDataUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.variants;
+
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.base.Splitter;
+
+public class VariantDataUtil {
+  private VariantDataUtil() {}
+
+  private static final Splitter DOT = Splitter.on(".");
+  private static final String ROOT = "$";
+
+  public static List<String> parsePath(String path) {
+    Preconditions.checkArgument(path != null, "Invalid path: null");
+    Preconditions.checkArgument(
```

Comment:
We probably want to check for `@` and possibly `?(` as those are common references used in json path.  It's best if we can reject as many unsupported paths as possible.

Reply:
These can only be used with a special meaning within brackets. I'll double check that these are allowed in JSON path though.

Reply:
Updated and added tests for this.

Reply:
I think we're also missing checks for `slice-selector` (see 2.3.4.1)